core/src/consensus/tendermint/stake/action_data.rs

```diff
@@ -150,4 +206,64 @@ mod tests {
         let account = StakeAccount::load_from_state(&state, &address).unwrap();
         assert_eq!(account.balance, 10);
     }
+
+    #[test]
+    fn stakeholders_track() {
+        let mut state = helpers::get_temp_state();
+        let addresses: Vec<_> = (1..100).map(|_| Address::random()).collect();
+        let accounts: Vec<_> = addresses
+            .iter()
+            .map(|address| StakeAccount {
+                address,
+                balance: thread_rng().gen_range(1, 100),
+            })
+            .collect();
+        {
+            let mut stakeholders = Stakeholders::load_from_state(&state).unwrap();
+            for account in &accounts {
+                stakeholders.update(account);
+            }
+            stakeholders.save_to_state(&mut state).unwrap();
+        }
+        let stakeholders = Stakeholders::load_from_state(&state).unwrap();
+        assert!(addresses.iter().all(|address| stakeholders.contains(address)));
+    }
+
+    #[test]
+    fn stakeholders_untrack() {
+        let mut state = helpers::get_temp_state();
+        let addresses: Vec<_> = (1..100).map(|_| Address::random()).collect();
+        let mut accounts: Vec<_> = addresses
+            .iter()
+            .map(|address| StakeAccount {
+                address,
+                balance: thread_rng().gen_range(1, 100),
+            })
+            .collect();
+        {
```

Comment:
Please do not abuse a block.
Block means the variables in the block must be dropped, not just scoping.

Reply:
Okay.